### PR TITLE
feat: add feature tour views without pagination component

### PR DESCRIPTION
**Pull request by siddarth2824**
## Problem
This PR adds the views, without the pagination component, for the product tour feature to display the functionality of the form builder features to new users.

## Solution

**Breaking Changes** 
- [x] No - this PR is backwards compatible  

**Features**:

- feat: add feature tour views without pagination component

**New dependencies**:

- `dependency` : [react-joyride](https://docs.react-joyride.com/) for feature tour 

## Before & After Screenshots
New story on the feature tour tooltip views without the pagination component added

### CODE REVIEW

File: frontend/src/features/admin-form/create/featureTour/FeatureTour.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    content: `${FEATURE_TOUR.SECOND_STEP.content}`,
+    disableBeacon: true,
+  },
+  {
+    target: `#${FEATURE_TOUR.THIRD_STEP.id}`,
+    title: `${FEATURE_TOUR.THIRD_STEP.title}`,
+    content: `${FEATURE_TOUR.THIRD_STEP.content}`,
+    disableBeacon: true,
+  },
+]
```
Comment (karrui) [inline frontend/src/features/admin-form/create/featureTour/FeatureTour.tsx:29]: a little weird for this constant to be created from another constant. can we not use this constant as the exported constant instead?

File: frontend/src/features/admin-form/create/featureTour/constants.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    content:
+      'Change your agency logo, theme colours, layout and add Instructions for your page',
+  },
+  THIRD_STEP: {
+    id: 'feature_tour_third_step',
+    title: 'Add Logic to your form',
+    content:
+      "Personalise your user's experience by showing fields and sections, or disabling submissions based on their input",
+  },
+}
```
Comment (karrui) [inline frontend/src/features/admin-form/create/featureTour/constants.ts:27]: Should we change this to an array instead of using a FeatureStep key? basically `FEATURE_TOUR[0] === FEATURE_TOUR.FIRST_STEP` anyways. 

then re: the above comment, we can reexport a `FEATURE_STEPS` constant using map like so:
```ts
export const FEATURE_STEPS = FEATURE_TOUR.map(({ id, ...rest}) => ({
  ...rest,
  target: `#${id}`,
  disableBeacon: true
}))
```

bonus if we can type this constant to joyride's step type.

File: frontend/src/features/admin-form/create/featureTour/utils.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      break
+    case 1:
+      marginTop = '8.5rem'
+      break
+    case 2:
+      marginTop = '12rem'
+      break
+    default:
+      marginTop = '5rem'
+  }
```
Comment (karrui) [inline frontend/src/features/admin-form/create/featureTour/utils.ts:17]: i suspect we do not need this weird margin prop. why is this needed? maybe discuss with designers too? seems like extra complexity for marginal gains.

File: frontend/src/features/admin-form/create/featureTour/utils.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+export const getFeaturTourToolTipMarginTop = (
```
Comment (karrui) [inline frontend/src/features/admin-form/create/featureTour/utils.ts:1]: nit: typo in name

File: frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        alignItems="center"
+        justifyContent="space-between"
+      >
+        <Text>Progress</Text>
+        {!isLastStep && (
+          <Button rightIcon={<BiRightArrowAlt />} {...primaryProps}>
+            Next
+          </Button>
+        )}
+        {isLastStep && <Button {...skipProps}>Done</Button>}
```
Comment (karrui) [inline frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.tsx:69]: nit: can be simplified to 

```ts
{ isLastStep ? next button : done button}
```

File: frontend/src/features/admin-form/create/featureTour/FeatureTour.tsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+export const FeatureTour = ({ shouldRun }: FeatureTourProps): JSX.Element => {
+  return (
+    <>
```
Comment (karrui) [inline frontend/src/features/admin-form/create/featureTour/FeatureTour.tsx:33]: nit: doesn't seem to need a fragment here

File: frontend/src/features/admin-form/create/featureTour/FeatureTour.tsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        floaterProps={{
+          placement: 'right',
+          styles: {
+            arrow: {
+              length: 8,
+              spread: 16,
+            },
+          },
+        }}
+        spotlightPadding={3}
```
Comment (karrui) [inline frontend/src/features/admin-form/create/featureTour/FeatureTour.tsx:47]: 3 seems to be a weird number, any reason? add comments?

File: frontend/src/features/admin-form/create/featureTour/FeatureTour.tsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        tooltipComponent={FeatureTourTooltip}
+        debug={true}
```
Comment (karrui) [inline frontend/src/features/admin-form/create/featureTour/FeatureTour.tsx:49]: i think we should turn debug off

File: frontend/src/features/admin-form/create/common/CreatePageSidebar/CreatePageSidebar.tsx [hunk 9]
```diff
@@ -49,18 +50,21 @@ export const CreatePageSidebar = (): JSX.Element | null => {
         icon={<BxsWidget fontSize="1.5rem" />}
         onClick={handleDrawerBuilderClick}
         isActive={activeTab === DrawerTabs.Builder}
+        id={FEATURE_TOUR.FIRST_STEP.id}
```
Comment (karrui) [inline frontend/src/features/admin-form/create/common/CreatePageSidebar/CreatePageSidebar.tsx:53]: if we use the array form, can easily replace this with
```ts 
id={FEATURE_TOUR[0].id}
```

**Review by karrui — CHANGES_REQUESTED** [review]

File: frontend/src/features/admin-form/create/featureTour/FeatureTour.tsx (hunk 7 above)
Comment (siddarth2824) [reply frontend/src/features/admin-form/create/featureTour/FeatureTour.tsx:47]: The react-joyride package has a prop to adjust the size of the spotlight on the targeted `DrawerIcon`. The default spotlight size is very big (size 10) and according to the Figma design, it seems to be much smaller than what the default spotlight size looks like. Hence I tried various numbers for the spotlight padding to adjust the size of the spotlight and it seems that 3 is the closest to the spotlight size in the Figma design.

File: frontend/src/features/admin-form/create/featureTour/utils.ts (hunk 3 above)
Comment (karrui) [reply frontend/src/features/admin-form/create/featureTour/utils.ts:17]: feels off, any reason why a custom tooltip will change the dynamic nature of the positioning? can you dig deeper into this?

Comment (siddarth2824) [reply frontend/src/features/admin-form/create/featureTour/utils.ts:17]: Solved it without using dynamic margins.

File: frontend/src/features/admin-form/create/featureTour/FeatureTour.tsx (hunk 7 above)
Comment (siddarth2824) [reply frontend/src/features/admin-form/create/featureTour/FeatureTour.tsx]: Attached below shows the type of spotlightPadding for the props that `Joyride` accepts.

Screenshot 2022-05-30 at 5 21 47 PM

File: frontend/src/features/admin-form/create/CreatePage.tsx [hunk 10]
```diff
@@ -1,13 +1,24 @@
...
 
 import { CreatePageContent } from './common/CreatePageContent'
 import { CreatePageSidebar } from './common/CreatePageSidebar'
 import { CreatePageSidebarProvider } from './common/CreatePageSidebarContext'
+import { FeatureTour } from './featureTour/FeatureTour'
 
 export const CreatePage = (): JSX.Element => {
+  const [startFeatureTour, setStartFeatureTour] = useState(false)
+
+  // need to add logic to only set startFeatureTour to be true if user
```
Comment (karrui) [inline frontend/src/features/admin-form/create/CreatePage.tsx:12]: nit: add `//TODO:` flag for easy search

File: frontend/src/features/admin-form/create/featureTour/constants.ts [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  {
+    id: 'feature_tour_third_step',
+    title: 'Add Logic to your form',
+    content:
+      "Personalise your user's experience by showing fields and sections, or disabling submissions based on their input",
+  },
+]
+
+export const FEATURE_STEPS = FEATURE_TOUR.map<Step>(
```
Comment (karrui) [inline frontend/src/features/admin-form/create/featureTour/constants.ts:29]: nit: put type on exported const instead of map?
```tsx
export const FEATURE_STEPS: Step[] = ...
```

File: frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.tsx [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+        {step.content}
+      </Text>
+      <Flex
+        flexDirection="row"
+        marginTop="2.5rem"
+        alignItems="center"
+        justifyContent="space-between"
+      >
+        <Text>Progress</Text>
+        {!isLastStep ? (
```
Comment (karrui) [inline frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.tsx:57]: nit: reverse conditional for readability?

**Review by karrui — COMMENTED** [review]
a couple nits. can i also have a storybook story so designers can vet? Can just show the tooltip as its own story np.

File: frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.tsx (hunk 12 above)
Comment (siddarth2824) [reply frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.tsx]: Updated code as per comments

File: frontend/src/features/admin-form/create/featureTour/constants.ts (hunk 11 above)
Comment (siddarth2824) [reply frontend/src/features/admin-form/create/featureTour/constants.ts]: Updated code as per comments

**Comment by karrui** [comment]
fix merge conflicts so chromatic CI/CD runs

File: frontend/src/features/admin-form/create/CreatePage.tsx (hunk 10 above)
Comment (siddarth2824) [reply frontend/src/features/admin-form/create/CreatePage.tsx]: Updated code as per comments

File: frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.tsx [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+interface FeatureTourStep {
+  content: React.ReactNode
+  title?: React.ReactNode
+}
+
+interface FeatureTourTooltipProps {
+  step: FeatureTourStep
+  tooltipProps: BoxProps
+  primaryProps: FeatureTourButtonProps
```
Comment (karrui) [inline frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.tsx:19]: nit: don't actually need a specific prop for this, you can also use ButtonProps, either ours or ChakraUI's

File: frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.stories.tsx [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+  onClick: () => {
+    return
+  },
+}
+
+const Template: Story = () => {
+  return (
+    <FeatureTourTooltip
+      step={mockStep}
+      tooltipProps={{} as BoxProps}
```
Comment (karrui) [inline frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.stories.tsx:33]: Don't think you need you cast here

File: frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.stories.tsx [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+      primaryProps={mockButtonProps}
```
Comment (karrui) [inline frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.stories.tsx:34]: Should not need any mocking once ButtonProp is used

File: frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.stories.tsx [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+      skipProps={mockButtonProps}
+      isLastStep={false}
+    />
+  )
+}
+
+export const BasicUsage = Template.bind({})
```
Comment (karrui) [inline frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.stories.tsx:41]: Can probably add more stories, such as if it is the last step.

Can even mock with a useState inside the template story to move between steps, and pass in an onclick handler to advance steps :)

File: frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.stories.tsx [hunk 17]
```diff
@@ -15,27 +20,37 @@ export default {
   },
 } as Meta
 
-const mockStep = {
-  content: `${FEATURE_TOUR[0].content}`,
-  title: `${FEATURE_TOUR[0].title}`,
-}
+const Template: Story<FeatureTourTooltipProps> = (args) => {
+  const [featureStep, setFeatureStep] = useState<number>(0)
```
[the remark this answers is not in the document]
Comment (siddarth2824) [reply frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.stories.tsx]: Updated as per comment

File: frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.stories.tsx (hunk 16 above)
Comment (siddarth2824) [reply frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.stories.tsx:57]: Updated as per comment

File: frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.stories.tsx (hunk 15 above)
Comment (siddarth2824) [reply frontend/src/features/admin-form/create/featureTour/FeatureTourTooltip.stories.tsx]: Updated as per comment
